File: Java/Leaderboard/src/main/java/tddmicroexercises/leaderboard/Driver.java
package tddmicroexercises.leaderboard;

public class Driver {

    private final String name;
    private final String country;

    Driver(String name, String country) {
        this.name = name;
        this.country = country;
    }

    String getName() {
        return name;
    }

    String getCountry() {
        return country;
    }

    @Override
    public int hashCode() {
        return name.hashCode() * 31 + country.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof Driver)) {
            return false;
        }
        Driver other = (Driver) obj;
        return this.name.equals(other.name) && this.country.equals(other.country);
    }

    @Override
    public String toString() {
        return name;
    }
}

File: kotlin/Leaderboard/src/main/kotlin/tddmicroexercises/leaderboard/Driver.kt
package tddmicroexercises.leaderboard

open class Driver(val name: String, val country: String) {

    override fun hashCode(): Int {
        return name.hashCode() * 31 + country.hashCode()
    }

    override fun equals(any: Any?): Boolean {
        if (this === any) {
            return true
        }
        if (any == null || any !is Driver) {
            return false
        }
        val other = any as Driver?
        return this.name == other!!.name && this.country == other.country
    }
}
